src/scheduling/schedule_service.py

```diff
@@ -143,6 +160,14 @@ def cleanup():
                     self._execution_service.cleanup_execution(execution_id, user)
 
                 self._execution_service.add_finish_listener(cleanup, execution_id)
+
+            if job.schedule.endOption == 'after':
+                job.schedule.endArg -= 1
```

Comment:
What do you think about not overriding the config file on every execution, but rather count how many executions could fit between now and start_datetime?
e.g. if `start_datetime=19:30`, `max_count=5`, `now=22:12`. Then until now there were 6 executions (19:30, 20:00, 20:30, 21:00, 21:30, 22:00) and there is no need to execute anymore? For this, we could reuse scheduling.schedule_config.ScheduleConfig.get_next_time

Reply:
Sorry, I'm not really understanding what you mean here as start is in the past? I think executions shouldn't be tied to exact next execute time as this makes this option obsolete in my opinion as this can be achieve with end time. This option would be more of  a guarantee execution of the job x times so even downtime wouldn't effect the job.

Reply:
Alright, makes sense. Another proposal: may be instead of overriding the maxCount, we could save one more parameter: executionsCount (i.e. how many times the script was executed). So that we will keep the original maxCount value as a source of truth

Reply:
Great idea, how about including executionsCount  on all repeatable tasks to offer some statistic about how many times the job ran in general.

Reply:
Sounds good to me

Reply:
Implemented.